module-telemetry/src/main/java/com/mapbox/maps/module/telemetry/MapTelemetryImpl.kt

```diff
@@ -17,50 +23,84 @@ class MapTelemetryImpl : MapTelemetry {
 
   private val appContext: Context
   private val accessToken: String
-  private val telemetry: MapboxTelemetry
+  private val eventsService: EventsServiceInterface
+  // We need telemetryService for location and metrics events which will be sent automatically if telemetryService is initialized.
+  private val telemetryService: TelemetryService
 
   /**
-   * Creates a map telemetry instance using appplication context and access token
+   * Creates a map telemetry instance using application context and access token
    *
    * @param appContext the application context
    * @param accessToken the mapbox access token
    */
   constructor(appContext: Context, accessToken: String) {
     this.appContext = appContext
     this.accessToken = accessToken
-    this.telemetry = MapboxTelemetry(appContext, accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT)
-    val telemetryState: TelemetryEnabler.State =
-      TelemetryEnabler.retrieveTelemetryStateFromPreferences()
-    if (TelemetryEnabler.State.ENABLED == telemetryState) {
-      telemetry.enable()
+
+    val eventsServiceOptions = EventsServerOptions(accessToken, BuildConfig.MAPBOX_EVENTS_USER_AGENT, null)
+    this.eventsService = EventsService.getOrCreate(eventsServiceOptions)
+
+    this.telemetryService = TelemetryService.getOrCreate(eventsServiceOptions)
+
+    if (TelemetryUtils.getEventsCollectionState()) {
+      enableTelemetryCollection(true)
```

Comment:
My point here was that `TelemetryUtils.getEventsCollectionState()` and `TelemetryUtils.setEventsCollectionState(...)` which happens inside `enableTelemetryCollection` gets/sets option to the same place. That's why it does not make sense to call 
```
if (TelemetryUtils.getEventsCollectionState()) {
      enableTelemetryCollection(true)
```
TelemetryUtils work with persistent settings and events service also checks `TelemetryUtils.getEventsCollectionState()` to determine if we need to send events. So no extra setting of events collection state is needed.

For MME we need this because event service did not check persistent settings for event collection state by itself.

Reply:
👍 do you suggest we should remove 
```
if (TelemetryUtils.getEventsCollectionState()) {
      enableTelemetryCollection(true)
}
```
all together?

Reply:
Yes. These get/set use the same place to get and set that's why this call does nothing. And I double-checked that everything works without this `enableTelemetryCollection` call: we get telemetry if it's enabled and we get turnstiles only if it's disabled.

Reply:
I also double checked, seems removing that logic doesn’t change the behaviour, will just remove it.